lib/client/renderer.js

```diff
@@ -490,29 +575,121 @@ function init (client, d3) {
             'marker-end':'url(#arrow)',
             'x1': chart().xScale(new Date(treatment.mills)),
             'y1': chart().yScale(client.sbx.scaleEntry(treatment)),
-            'x2': d3.event.x - this.offsetWidth/2,
-            'y2': chart().yScale(client.sbx.scaleEntry(treatment)),
+            'x2': x,
+            'y2': y,
             'stroke-width': 2,
             'stroke': 'white'
           });
+          
       })
       .on('dragend', function() { 
+        var dboperation;
+        chart().drag.selectAll('.drag-droparea').remove();
         hideTooltip();
-        if (window.confirm(translate('Change treatment time to %1 ?', { params: [newTime.toLocaleTimeString()] } ))) {
-          client.socket.emit(
-            'dbUpdate', 
-            { 
-              collection: 'treatments', 
-              _id: treatment._id,
-              data: { created_at: newTime.toISOString() } 
-            },
-            function callback(result) { 
-              console.log(result); 
-              chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+        switch (operation) {
+          case "Move":
+            if (window.confirm(translate('Change treatment time to %1 ?', { params: [newTime.toLocaleTimeString()] } ))) {
+              client.socket.emit(
+                'dbUpdate', 
+                { 
+                  collection: 'treatments', 
+                  _id: treatment._id,
+                  data: { created_at: newTime.toISOString() } 
+                },
+                function callback(result) { 
+                  console.log(result); 
+                  chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+                }
+              );
+            } else {
+              chart().drag.selectAll('.arrow').remove();
             }
-          );
-        } else {
-          chart().drag.selectAll('.arrow').remove();
+            break;
+          case "Remove":
+            if (window.confirm(translate('Remove treatment ?'))) {
+              client.socket.emit(
+                'dbRemove', 
+                { 
+                  collection: 'treatments', 
+                  _id: treatment._id
+                },
+                function callback(result) { 
+                  console.log(result); 
+                  chart().drag.selectAll('.arrow').transition().duration(5000).style('opacity', 0).remove();
+                }
+              );
+            } else {
+              chart().drag.selectAll('.arrow').remove();
+            }
+            break;
+          case "Move insulin":
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1849529815/issues/source?bid=2452379&fileBranchId=3084282#l625)